### PR TITLE
Enhance header filtering in getHeadersApplicableToAllResources function to exclude falsy values

### PR DESCRIPTION
**Pull request by ivanvakulov**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This pull request includes a small change to the `src/utils/headers.ts` file. The change ensures that only headers with non-falsy values are included in the result.

* [`src/utils/headers.ts`](diffhunk://#diff-0dc4e26c4023e2ae5951bcb26642a06be079e6361e99c0f5dd4bd70fac43e71bR180): Added a filter to remove headers with falsy values in the `getHeadersApplicableToAllResources` function.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes (if not applicable, please state why)

### CODE REVIEW

File: src/utils/headers.ts [hunk 1]
```diff
@@ -177,6 +177,7 @@ export function getHeadersApplicableToAllResources(headers: SecurityHeaders) {
     Object.entries(headers)
     .filter(([key]) => appliesToAllResources(key as OptionKey))
     .map(([key, value]) => ([getNameFromKey(key as OptionKey), headerStringFromObject(key as OptionKey, value)]))
+    .filter(([_, value]) => Boolean(value))
```
Comment (vejja) [inline src/utils/headers.ts:180]:
```suggestion
    .filter(([, value]) => Boolean(value))
```

Addresses CodeQL warning

**Review by vejja — CHANGES_REQUESTED** [review]

**Comment by ivanvakulov** [comment]
@vejja Thank you for the review. I've added new tests for publicAssets to ensure there are no empty headers and that headers are correctly set according to the routeRules.

**Review by vejja — APPROVED** [review]

**Comment by vejja** [comment]
@Baroshem LGTM

**Review by Baroshem — APPROVED** [review]
Amazing work @ivanvakulov 

Thanks @vejja for checking it!

**Pull request merged**